### ISSUE TITLE
Various labels and inputs not associated with each other on timecard page

### ISSUE THREAD

**Issue by maya** [18F/tock#533]
Various labels and inputs (selects and textarea) are not associated with each other. They should be the same so they're explicitly associated with each other per W3C F68 4.1.2 https://www.w3.org/TR/WCAG20-TECHS/F68. In all cases, the ids are different (as they should be) and the "for"s are all the same (should not be).

They are the following:
### 1. Project select

```
<label for="project">Project:</label>
<select id="id_timecardobject_set-0-project" name="timecardobject_set-0-project" style="display: none;">...</select>
```
### 2. Hours input

```
<div class="entry-amount">
  <label for="hours">Hours Spent:</
  <input id="id_timecardobject_set-0-hours_spent" min="0" name="timecardobject_set-0-hours_spent" step="0.25" type="number">
</div>
```
### 3. General snippet textarea

```
<div class="entry-notes" id="entry-notes">
  <label for="notes">Notes:</label>
  Please provide a snippet about how you spent your time.
  <textarea class="entry-notes-text" cols="40" id="id_timecardobject_set-1-notes" name="timecardobject_set-1-notes" rows="10"></textarea>
</div>
```
### 4. Delete entry checkbox

```
<div class="entry-delete">
  <input id="id_timecardobject_set-0-DELETE" name="timecardobject_set-0-DELETE" type="checkbox">
</div>
```

*Does not have labels, when added needs to match.

**Comment by maya** [comment]
Closed via #523.
